boba/cli.py

```diff
@@ -18,14 +18,19 @@
               default='.', show_default=True)
 @click.option('--lang', help='Language, can be python/R [default: inferred from file extension]',
               default='')
-def compile(script, out, lang):
+@click.option('--decisions', '-d', cls=PythonDict, default='{}')
+def compile(script, out, lang, decisions):
```

Comment:
Would it be more flexible if we add this "decisions" option to `boba run` instead of `boba compile`? So we only need to generate the entire multiverse once, and we could have the flexibility of running a small subset based on decision values. This will be a very useful feature to add to `boba run`.

Reply:
And the implementation will be much simpler as we'll only need to read and filter the `summary.csv` to get the correct universe ids. Are there particular reasons that modifying `boba compile` is better?

Reply:
the main issue with making this part of `boba run` and not `boba compile` is continuous variables.
Let's say i have a template that has a continuous variable `A` with range `0..1`, and a static variable `B` with options `"red", "green", "blue"`.
Now lets look at the two different cases, filter on `boba run` and filter on `boba compile`.

filter on `boba run`:
We already ran `boba compile` and generated the universes. This means there is some number of universes where `A` is in the range `0..1` and `B` has values `"red", "green", "blue"`.
Filtering on `B` is fine: we can easily do `boba run -d "{'B': 'red'}"`, and that would run all of the generated universes that have B = "red".
Filtering on `A`, however, is problematic. Suppose I do `boba run -d "{'A': 0.5}"`. Per the spec provided by our user, this is legal. `A` should be able to take on any value in the range `0..1`. But in the implementation, this is not the case, as `boba compile` creates a set of discrete universes via sampling, not infinite universes. Thus, there may not be a generated universe such that A = 0.5.
How could we fix this? Well, we could pick the universe with the closest `A` value, but that is technically lying to the user since we're not actually testing with A = 0.5. And what if the number of samples is low and there really isn't a 'close enough' A?

filter on `boba compile`:
Again, filtering on `B` is trivial: simply generate the universes where `B` is a particular value.
Filtering on `A` is also trivial now, since we can simply generate the universes where `A` is a particular value.

That was my reasoning behind choosing to use `boba compile` instead of `boba run`.

Reply:
Thanks! The example was helpful and I think the reasoning makes sense. As we discussed in the meeting, besides the review points we'll also need to combine `path_filter` with the rest, as placeholder vars and blocks aren't allowed to have the same identifier.